R/Execution.R

```diff
@@ -101,149 +91,47 @@ execute <- function(analysisSpecifications,
-      # however, working out how to handle dependencies wasn't obvious
-      # This approach could be modified to allow multiple executionSettings, but that would require a substantial re-write
-      for (i in 1:length(analysisSpecificationsLoad$moduleSpecifications)) {
-        moduleSpecification <- analysisSpecificationsLoad$moduleSpecifications[[i]]
-        targetName <- sprintf("%s_%d", moduleSpecification$module, i)
-        dependencyModules <- dependencies[dependencies$module == moduleSpecification$module, ]$dependsOn
-        dependencyTargetNames <- moduleToTargetNames[moduleToTargetNames$module %in% dependencyModules, ]$targetName
-
-        # Use of tar_target_raw allows dynamic names
-        targetList[[length(targetList) + 1]] <- targets::tar_target_raw(targetName,
-          substitute(Strategus:::runModule(analysisSpecifications, keyringSettings, i, executionSettings),
-            env = list(i = i)
-          ),
-          deps = c("analysisSpecifications", "keyringSettings", "executionSettings", dependencyTargetNames)
-        )
-
-        if (execResultsUpload) {
-          resultsTargetName <- paste0(targetName, "_results_upload")
-          targetList[[length(targetList) + 1]] <- targets::tar_target_raw(resultsTargetName,
-            substitute(Strategus:::runResultsUpload(analysisSpecifications, keyringSettings, i, executionSettings),
-              env = list(i = i)
-            ),
-            deps = c("analysisSpecifications", "keyringSettings", "executionSettings", targetName)
-          )
-        }
-      }
-      targetList
-    },
-    script = fileName
-  )
-
-  # Store settings objects in the temp folder so they are available in targets
-  analysisSpecificationsFileName <- .formatAndNormalizeFilePathForScript(file.path(executionScriptFolder, "analysisSpecifications.rds"))
-  saveRDS(analysisSpecifications, analysisSpecificationsFileName)
-  executionSettingsFileName <- .formatAndNormalizeFilePathForScript(file.path(executionScriptFolder, "executionSettings.rds"))
-  saveRDS(executionSettings, executionSettingsFileName)
-  keyringSettingsFileName <- .formatAndNormalizeFilePathForScript(file.path(executionScriptFolder, "keyringSettings.rds"))
-  saveRDS(list(keyringName = keyringName), keyringSettingsFileName)
-
-  # Generate target names by module type
-  moduleToTargetNames <- list()
+  # Execute any other modules
   for (i in 1:length(analysisSpecifications$moduleSpecifications)) {
-    moduleSpecification <- analysisSpecifications$moduleSpecifications[[i]]
-    targetName <- sprintf("%s_%d", moduleSpecification$module, i)
-    moduleToTargetNames[[length(moduleToTargetNames) + 1]] <- tibble(
-      module = moduleSpecification$module,
-      targetName = targetName
-    )
+    moduleName <- analysisSpecifications$moduleSpecifications[[i]]$module
+    if (tolower(moduleName) != "cohortgeneratormodule") {
+      moduleObj <- get(moduleName)$new()
+      moduleObj$execute(
```

Comment:
Would be useful to have this in a tryCatch() so that we can continue to make progress on the execution/upload in the case of a failure.

Reply:
Linking this to #141 as we'll address this in a separate feature branch for this topic.